pkg/operator/sync.go

```diff
@@ -850,17 +856,26 @@ func (optr *Operator) waitForControllerConfigToBeCompleted(resource *mcfgv1.Cont
 	return nil
 }
 
-func (optr *Operator) getOsImageURL(namespace string) (string, error) {
+// getOsImageURLs returns (new type, old type) for operating system update images.
+func (optr *Operator) getOsImageURLs(namespace string) (string, string, error) {
 	cm, err := optr.mcoCmLister.ConfigMaps(namespace).Get(osImageConfigMapName)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	releaseVersion := cm.Data["releaseVersion"]
 	optrVersion, _ := optr.vStore.Get("operator")
 	if releaseVersion != optrVersion {
-		return "", fmt.Errorf("refusing to read osImageURL version %q, operator version %q", releaseVersion, optrVersion)
+		return "", "", fmt.Errorf("refusing to read osImageURL version %q, operator version %q", releaseVersion, optrVersion)
+	}
+	newformat, ok := cm.Data["baseOperatingSystemContainer"]
+	if !ok {
+		return "", "", fmt.Errorf("Missing baseOperatingSystemContainer from configmap")
```

Comment:
Is `baseOperatingSystemContainer` mandatory to have? On a non-layered system is it just empty (but the key exists?)